packages/admin/admin/src/dataGrid/usePersistentColumnState.tsx

```diff
@@ -1,17 +1,100 @@
+import { Breakpoint, useTheme } from "@mui/material";
 import { DataGridProps, GridColumnVisibilityModel, useGridApiRef } from "@mui/x-data-grid";
 import * as React from "react";
 
+import { useWindowSize } from "../helpers/useWindowSize";
 import { useStoredState } from "../hooks/useStoredState";
+import { GridColDef, GridColView } from "./GridColDef";
 
-export function usePersistentColumnState(stateKey: string): Omit<DataGridProps, "rows" | "columns"> {
+type UserVisibilityChange = {
+    column: string;
+    visible: boolean;
+    view: GridColView;
+};
+
+const getUpdatedUserVisibilityChanges = (
+    existingUserChanges: UserVisibilityChange[],
+    newColumnVisibilities: GridColumnVisibilityModel,
+    currentView: GridColView,
+): UserVisibilityChange[] => {
+    let updatedUserChanges = [...existingUserChanges];
+
+    Object.keys(newColumnVisibilities).forEach((changedColumn) => {
+        const newVisible = newColumnVisibilities[changedColumn];
+
+        updatedUserChanges = updatedUserChanges.filter((change) => {
+            return change.column !== changedColumn || change.view !== currentView;
+        });
+
+        updatedUserChanges.push({
+            column: changedColumn,
+            visible: newVisible,
+            view: currentView,
+        });
+    });
+
+    return updatedUserChanges;
+};
+
+const useGridColumns = (apiRef: ReturnType<typeof useGridApiRef>) => {
+    const [columns, setColumns] = React.useState<GridColDef[] | undefined>();
+
+    React.useEffect(() => {
+        // This will be `undefined` if the free version of DataGrid V5 is used.
+        setColumns(apiRef.current?.getAllColumns?.());
+    }, [apiRef]);
```

Comment:
Should we add a warning here during development?

Reply:
Do you mean a `console.warn` or something else?

Before doing things explicitly for the old version, I think we should just update to V7 😄